src/apps/vpn/mozillavpn.cpp

```diff
@@ -897,6 +897,8 @@ void MozillaVPN::mainWindowLoaded() {
 #endif
 #ifdef SENTRY_ENABLED
   SentryAdapter::instance()->init();
+  QObject::connect(controller(), &Controller::readyToQuit,
+                   SentryAdapter::instance(), &SentryAdapter::onBeforeShutdown);
```

Comment:
Isn't it possible to move this into SentryAdapter::init() for encapsulation? Also, a comment may be useful about why this is being done when Controller::readyToQuit is emitted.

Reply:
Not right now, however this can wait until this is merged:
https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7520

Right now Sentry-Adapter sit's in the "shared" part, therefore it cannot know about the Controller.